Source/Core/Core/HW/Memmap.h

```diff
@@ -37,23 +37,21 @@ extern u8* m_pEXRAM;
 extern u8* m_pL1Cache;
 extern u8* m_pFakeVMEM;
 
-enum
-{
-  // RAM_SIZE is the amount allocated by the emulator, whereas REALRAM_SIZE is
-  // what will be reported in lowmem, and thus used by emulated software.
-  // Note: Writing to lowmem is done by IPL. If using retail IPL, it will
-  // always be set to 24MB.
-  REALRAM_SIZE = 0x01800000,
-  RAM_SIZE = MathUtil::NextPowerOf2(REALRAM_SIZE),
-  RAM_MASK = RAM_SIZE - 1,
-  FAKEVMEM_SIZE = 0x02000000,
-  FAKEVMEM_MASK = FAKEVMEM_SIZE - 1,
-  L1_CACHE_SIZE = 0x00040000,
-  L1_CACHE_MASK = L1_CACHE_SIZE - 1,
-  IO_SIZE = 0x00010000,
-  EXRAM_SIZE = 0x04000000,
-  EXRAM_MASK = EXRAM_SIZE - 1,
-};
+// g_ram_size is the amount allocated by the emulator, whereas g_ram_size_real is
+// what will be reported in lowmem, and thus used by emulated software.
+// Note: Writing to lowmem is done by IPL. If using retail IPL, it will
+// always be set to 24MB.
+extern u32 g_ram_size_real;
```

Comment:
I meant to bring this up in the previous review, but these should likely be hidden from external view and be static file-scope variables and instead have functions that return by value to query the values of these variables. This way other subsystems can query, but not modify these values unintentionally.

Reply:
This still applies, just to make sure it isn't forgotten about.

Reply:
I'm not a fan of the disconnect between the actual 11 variables and the 11 query functions.  I'd say just don't write bad code, but if this is the last thing keeping you...

Reply:
Most of these aren't even read outside of Memmap.cpp as far as I can tell so you don't need to bother exposing them at all.

Reply:
> Most of these aren't even read outside of Memmap.cpp as far as I can tell so you don't need to bother exposing them at all.

This.

Historically, global variables have been a massive pain in Dolphin and a source of frustration for refactoring. If they don't need to be exposed, then they shouldn't be. Especially if they're mutable global variables. It turns a "you shouldn't do that" kind of mistake vector into a "you cannot do that". 

"just don't write bad code", is (IMO) a terrible viewpoint to have especially when the language already provides the necessary facilities to allow one to completely eliminate such code from being written in the first place directly at the interface boundary.